app/models/classroom.rb

```diff
@@ -1,6 +1,8 @@
 class Classroom < ApplicationRecord
   belongs_to :user
-  has_many :participations
-  has_many :students, class_name: :user, through: :participations, source: :user
+  has_many :participations, dependent: :destroy
+  has_many :students, through: :participations, source: :user
```

Comment:
since you removed the class name, can other functions in other models still work? As you only kept students, but we did not reference anywhere the a standard user is called a student

Reply:
So, we had this issue where Rails got a bit confused about who our students are in the Classroom. It was looking for a class called Student, but we actually have a class called User.

So, to clear up the confusion, I told Rails, "Hey, those students are actually Users!" Now everything should work smoothly without any head-scratching from the system   

here is correct line for that ➡️ :  has_many :students, class_name: 'User', through: :participations, source: :user
I can adjust it in my code or we can change it in right here in GitHub